### PR TITLE
docs: Add test specifications for `TokenPauseTransaction`

### LINKED ISSUE #166: Document E2E tests: `TokenPauseTransaction`
Document the E2E `TokenPauseTransaction` tests in `test-specifications/`.

### PR DESCRIPTION
**Pull request by rwalworth**
**Description:**
This PR adds the test specifications for `TokenPauseTransaction` fields.

**Related Issue(s):**

Fixes #166

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ivaylogarnev-limechain — APPROVED** [review]
LGTM!

**Comment by jsync-swirlds** [comment]
Would it be a good idea to add tests here for the various affected token transactions?
Something like verifying that a tokenTransfer fails if the token is paused?

**Comment by rwalworth** [comment]
> Would it be a good idea to add tests here for the various affected token transactions? Something like verifying that a tokenTransfer fails if the token is paused?

That will be a case tested when the tests are written for `TransferTransaction`.

File: test-specifications/token-service/tokenPauseTransaction.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+| 4       | Pauses a token that doesn't exist                         | tokenId="123.456.789"                                                                   | The token pause fails with an INVALID_TOKEN_ID response code from the network.  | N                 |
+| 5       | Pauses a token that is deleted                            | tokenId=<DELETED_TOKEN_ID>, commonTransactionParams.signers=[<DELETED_TOKEN_PAUSE_KEY>] | The token pause fails with an TOKEN_WAS_DELETED response code from the network. | N                 |
+| 6       | Pauses a token that is empty                              | tokenId=""                                                                              | The token pause fails with an SDK internal error.                               | N                 |
+| 7       | Pauses a token twice                                      | tokenId=<CREATED_TOKEN_ID>, commonTransactionParams.signers=[<CREATED_TOKEN_PAUSE_KEY>] | The token pause fails with an TOKEN_IS_PAUSED response code from the network.   | N                 |
+
```
Comment (0xivanov) [inline test-specifications/token-service/tokenPauseTransaction.md:59]: Shall we also add a scenario where the transaction is signed with the admin key?

Comment (rwalworth) [reply test-specifications/token-service/tokenPauseTransaction.md:59]: You mean instead of the pause key? Is that allowed behavior?

Comment (jsync-swirlds) [reply test-specifications/token-service/tokenPauseTransaction.md:59]: Admin key cannot unpause; only the pause key can do that.  Admin key can change the keys (if the key is not disabled), and a few other tasks, but any task with its own key is denied to admin and dedicated solely to that key.

A test signed with admin and _expected to fail_ might be useful.

Comment (rwalworth) [reply test-specifications/token-service/tokenPauseTransaction.md:60]: Gotchya, added a test

**Review by 0xivanov — APPROVED** [review]
LGTM

**Pull request merged**